tests/integration/test_dcim.py

```diff
@@ -17,115 +17,137 @@ class TestSimpleServerRackingAndConnecting:
     """Verify we can create, rack, and connect a server."""
 
     @pytest.fixture
-    def site(self, nb_client):
-        """Verify we can create a site."""
-        site_name = "MSP"
-        site = nb_client.dcim.sites.create(name=site_name, slug=site_name.lower().replace(" ", "_"), status="active")
-        assert site
+    def location(self, nb_client):
+        """Verify we can create a location."""
+        location_types = ["dcim.device", "dcim.rack", "dcim.rackgroup"]
+        site_type = nb_client.dcim.location_types.create(name="Site", content_types=location_types, nestable=True)
+        location_name = "MSP"
+        location = nb_client.dcim.locations.create(
+            name=location_name,
+            slug=location_name.lower().replace(" ", "_"),
+            status={"name": "Active"},
+            location_type=site_type.id,
+            depth=3,
+        )
+        assert location
+
+        return location
+
+    @pytest.fixture
+    def rack_group(self, location):
+        """Verify we can create a rack device."""
+        rack_group = location.api.dcim.rack_groups.create(name="rack_group_1", location=location.id)
+        assert rack_group
 
-        return site
+        return rack_group
 
     @pytest.fixture
-    def rack(self, site):
+    def rack(self, location, rack_group):
         """Verify we can create a rack device."""
-        rack = site.api.dcim.racks.create(name="rack1", site=site.id, status="active")
+        rack = location.api.dcim.racks.create(
+            name="rack1", location=location.id, rack_group=rack_group.id, status={"name": "Active"}
+        )
         assert rack
 
         return rack
 
     @pytest.fixture
-    def data_leafs(self, rack):
+    def data_leafs(self, rack, location):
         """Verify we can create data leaf switch devices."""
         devices = []
         for i in [1, 2]:
             device = rack.api.dcim.devices.create(
                 name=f"access_switch{i}.networktocode.com",
                 device_type={"slug": "dcs-7050tx3-48c8"},
```

Comment:
This needs to change probably to a `model` reference instead of `slug`. Same comment below.